src/dialog-editor/components/modal/modalComponent.ts

```diff
@@ -0,0 +1,304 @@
+      box: this.DialogEditor.getDialogTabs()[
+        this.DialogEditor.activeTab].dialog_groups[
+          this.elementInfo.boxId],
+      field: this.DialogEditor.getDialogTabs()[
+        this.DialogEditor.activeTab].dialog_groups[
+          this.elementInfo.boxId].dialog_fields[
+            this.elementInfo.fieldId]
+    };
+    return this.elementInfo.type in elements &&
+      _.isMatch(elements[this.elementInfo.type], this.modalData);
+  }
+
+  /**
+   * Store modified data back to the service.
+   * @memberof ModalController
+   * @function saveDialogFieldDetails
+   */
+  public saveDialogFieldDetails() {
+    switch (this.elementInfo.type) {
+      case 'tab':
+        _.extend(
+          this.DialogEditor.getDialogTabs()[
+            this.DialogEditor.activeTab],
+          { label: this.modalData.label,
+            description: this.modalData.description }
+        );
+        break;
+      case 'box':
+        _.extend(
+          this.DialogEditor.getDialogTabs()[
+            this.DialogEditor.activeTab].dialog_groups[
+              this.elementInfo.boxId],
+          { label: this.modalData.label,
+            description: this.modalData.description }
+        );
+        break;
+      case 'field':
+        this.DialogEditor.getDialogTabs()[
+          this.DialogEditor.activeTab].dialog_groups[
+            this.elementInfo.boxId].dialog_fields[
+              this.elementInfo.fieldId] = this.modalData;
+        break;
+      default:
+        break;
+    }
+  }
+
+  /**
+   * Delete dialog field selected in modal.
+   * @memberof ModalController
+   * @function deleteField
+   */
+  public deleteField() {
+    _.remove(
+      this.DialogEditor.getDialogTabs()[
+        this.DialogEditor.activeTab
+      ].dialog_groups[
+        this.elementInfo.boxId
+      ].dialog_fields,
+      (field: any) => field.position === this.elementInfo.fieldId
+    );
+  }
+
+  /**
+   * Add entry for radio button / dropdown select.
+   * @memberof ModalFieldController
+   * @function addEntry
+   */
+  public addEntry() {
+    this.modalData.values.push(['', '']);
+  }
+
+  /**
+   * Remove entry for radio button / dropdown select
+   * @memberof ModalFieldController
+   * @function removeEntry
+   * @param entry to remove from array
+   */
+  public removeEntry(entry: any) {
+    _.pull(this.modalData.values, entry);
+  }
+
+  /**
+   * Finds entries for the selected category.
+   * @memberof ModalController
+   * @function currentCategoryEntries
+   */
+  public currentCategoryEntries() {
+    if (ng.isDefined(this.categories)) {
+      return _.find(
+        this.categories.resources,
+        'id',
+        this.modalData.options.category_id
+      );
+    }
+  }
+
+  public showModal(options: any) {
+    options.controller = ['parent', function(parent) { this.parent = parent; }];
+    options.resolve = {
+      parent: () => this
+    };
+    options.controllerAs = 'modalCtrl';
+    options.template = `<${options.component}
```

Comment:
```
I think this would be more readable like this:
options.template = `
<${options.component}
  modal-data="modalCtrl.parent.modalData"
  element-info="modalCtrl.parent.elementInfo"
  categories="modalCtrl.parent.categories"
  add-entry="modalCtrl.parent.addEntry"
  remove-entry="modalCtrl.parent.removeEntry"
  current-category-entries="modalCtrl.parent.currentCategoryEntries"
  resolve-categories="modalCtrl.parent.resolveCategories"
  modal-tab-is-set="modalCtrl.parent.modalTabIsSet"
  modal-tab-set="modalCtrl.parent.modalTabSet"
  modal-tab="modalCtrl.parent.modalTab"
  save-modal="modalCtrl.parent.saveDialogFieldDetails"
  uib-modal-instance="modalCtrl.parent.uibModalInstance"
></${options.component}>
`;
```